src/sql/parts/notebook/models/contracts.ts

```diff
@@ -0,0 +1,47 @@
+
```

Comment:
This class is essentially being ported as-is with changes just to recompile against internal vs. extension APIs. May need refresh since we took some small changes in the extension. It's taking what are now internal interfaces and moving to a sensible location